src/librustc/mir/interpret/error.rs

```diff
@@ -113,6 +113,7 @@ pub enum EvalErrorKind<'tcx, O> {
     Layout(layout::LayoutError<'tcx>),
     HeapAllocZeroBytes,
     HeapAllocNonPowerOfTwoAlignment(u64),
+    ForeignStatic,
```

Comment:
This should probably be `ReadFromForeignStatic`.

Reply:
No, that's why we can't error in const_qualif - borrowing a foreign static is perfectly fine.

Reply:
Oh yes, duh.